frontend/public/components/app.jsx

```diff
@@ -275,7 +275,11 @@ if ('serviceWorker' in navigator) {
   }
 }
 
-const AppGuard = (props) => <AsyncComponent loader={() => coFetch('api/tectonic/version').then(() => App)} {...props} />;
+const AppGuard = connect(({k8s}) => ({inFlight: k8s.getIn(['RESOURCES', 'inFlight'])}))(
```

Comment:
This will block until API discovery is complete?

Reply:
Yes, we have to because we start with 0 models registered and will get errors when we try and `fetch` namespaces for the dropdown.

Reply:
Yeah, I figured that's the case. It's a shame because sometimes discovery can be slow. We blocked in OpenShift console during discovery, and it was noticeable.

Reply:
We could add an "app shell" that doesn't depend on API calls and increases perceived loading time.

Reply:
After using this branch on a slower network, this is definitely not a good UX. I'll try to find a way to not block the UI on API discovery.